### ISSUE TITLE
CaRL version 1.1

### ISSUE THREAD

**Issue by Trunk2004** [autonomousvision/CaRL#11]
Hi @Kait0 , regarding about the CaRL 1.1 version does the training script only run with the CARLA custom build as with the same script the process in 1.1 cause the leaderboard end after checking cuda available and therefore stop the training and the parent script kill all and repeat.

Then i clone the initial 1.0 release and using the same script, only then the training is running. So does the issue that do i need to custom build CARLA to be able to run with the change in 1.1?

**Comment by Kait0** [comment]
I would not train with only 1 env. Use at least 6-8.
PPO will get very correlated gradients (your data is correlated in time, but stochastic gradient descent assumes they are uncorrelated samples). Multi-env is important in RL to decorrelate gradients within your mini-batch.

**Comment by Trunk2004** [comment]
Hi, do you still keep the output log of your training and can you share it with me ? So I can figure something wrong with my  training.
If not, can you take a look at my output log ?The episodic return is decline overtime and show no sign of recover from agent exploration.
This is the output log of wandb:
[output.log](https://github.com/user-attachments/files/23137158/output.log)

This is the config i use for training:
[config.json](https://github.com/user-attachments/files/23137168/config.json)

Any advice will be helpful. Thank you.

**Comment by Trunk2004** [comment]
Thanks, look like for a already limited hardware like mine, the camara sensors is not a good way to train. I want to ask that it is possible to adapt other reinforcement algorithms from cleanrl into the CaRL like PPO ?

**Comment by Kait0** [comment]
Well, I implemented the PPO algorithm based on cleanrl, so you can also implement other algos based on that repo.

If you do an evaluation/benchmarking, you shouldn't change the timeout / blocked value since that would be modifying the metric (which has blocked and timeout penalties based on the preset thresholds).

The despawning and respawning of the surrounding traffic is part of how the leaderboard sets up scenarios.
Longest6 v2 is autoconverted from longest6 (leaderboard 1.0), not all scenarios could be converted, so yes, it's normal for some of them not to set up.

Metrics are computed per route. When all routes are done, you need to run the [aggregation_script](https://github.com/autonomousvision/CaRL/tree/main/CARLA#result-aggregation).

Not sure what you mean by coping. There is no speed limit penalty in the metric; compliance is optional (but we do include it in the reward).

CARLA has bicycles and motorcycles. Think the standard leaderboard background traffic doesn't use them outside of the scenarios (but there is 1 scenario where a bike can cross the street I think).

**Comment by Trunk2004** [comment]
> If you are Gpu memory constraint you can try reducing total batch size -> 2048/ 1024 or 1024 / 1024.

So I tried your suggestion and even at 1024 / 1024 it still have chance to cause a OOM. So i may tried cutting down to 512/512 but I want to ask that with 6 env it was 512/512 then if I reduce the num env, I can increase the batch size till OOM like your suggestions?
Thanks

**Comment by Trunk2004** [comment]
> Maybe make it divisible by the number of envs you have.

Hi, @Kait0 , why you suggest the mini batch size to be divisible by the number of envs? It is related to how the transitions is collected through multiple envs per step?

**Comment by Kait0** [comment]
otherwise the mini-batch will just be padded with copied samples. won't have much of an impact but makes most sense
